compiler/tools/unicodetables.d

```diff
@@ -0,0 +1,208 @@
+/**
+Generates the Unicode tables and associated Identifier tables for dmd-fe.
+
+You should ideally use ldc2 for running this program.
+Compiled with dmd you will be waiting a while due to ``ValueRanges.merge``.
+
+I.e. ``ldc2 -O3 -i -run unicodetables.d``
```

Comment:
What does the program generate?

Reply:
Currently: ``dmd.common.identifiertables.d``.

Which is C99, C11, UAX31 start/continue, and least restrictive set aka all start/continue. 

We can add other fixed tables for other C versions if we need to pretty easily. https://github.com/dlang/dmd/pull/15307/files#diff-dac79f437668fde367bf8cb9b5ba115de54ea04139f1e960c1d952490d305700
https://github.com/dlang/dmd/pull/15307/files#diff-2b623fcefb34769f44354e868a18eea9c052aa6a1fab8eafc67774a6582cc10aR136

I split it up into quite a few modules, due to it needing to generate tables for things like normalization later on.

Reply:
Thank you for the explanation. I wasn't clear, though. The explanation should be added to the code.

I presume that generating the tables will not be run when the compiler is rebuilt?

Reply:
Oh goodness no.

We'll only need to rerun it when Unicode updates. We can be pretty lazy about doing it.

As long as it's always committed in a working state we're ok, it's all pretty naive code that should last a long time. It should also be updated to have the Unicode version in the string. Same policy as Phobos's table generator.